epfl-menus.php

```diff
@@ -1558,13 +1559,40 @@ static private function _get_publish_controller ($menu) {
      * Shall be called whenever $menu changes (from the point
      * of view of @link get_menu)
      */
-    static function menu_changed ($menu, $causality = NULL) {
-        $publisher = static::_get_publish_controller($menu);
+    static function menu_changed ($menu) {
+        $site_url = trim(get_site_url());
+        $site_url = trailingslashit( $site_url );
+        if (substr($site_url, -1) === '/') {
+            $site_url = substr($site_url, 0, -1);
+        }
 
-        if ($causality) {
-            $publisher->forward($causality);
+        if (function_exists('pll_get_term_language')) {
+            $menu_language =  pll_get_term_language($menu->get_term_id());
         } else {
-            $publisher->initiate();
+            $menu_language = 'en';
+        }
+
+        $host = get_menu_api_base_url();
+        $url_api = "{$host}/refreshSingleMenu/?url={$site_url}"; # TODO manage language:  &lang=' . $menu_language
+
+        $curl = curl_init($url_api);
+        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($curl, CURLOPT_CONNECTTIMEOUT, getenv('MENU_API_CURLOPT_CONNECTTIMEOUT') ?? 1);
+        curl_setopt($curl, CURLOPT_TIMEOUT, getenv('MENU_API_CURLOPT_TIMEOUT') ?? 1);
+        $response = curl_exec($curl);
+        if (curl_errno($curl)) {
+            $error_text = curl_error($curl);
+        }
+        curl_close($curl);
+
+        if (isset($error_text)) {
+            error_log( "curl error: {$error_text} at {$url_api}" );
+            return NULL;
```

Comment:
Please throw an exception (even though there is little point in managing errors correctly in a `shutdown` hook)